### ISSUE TITLE
≥3 Axes overlap in layered plots by default

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#2634]
<img width="301" alt="screen shot 2017-07-06 at 10 25 35" src="https://user-images.githubusercontent.com/589034/27924264-84d5924c-6235-11e7-9436-0aa424fd0dc9.png">

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "width": 460,
  "height": 240,
  "layer": [
    {
      "data": {
        "url": "data/co2.csv"
      },
      "mark": "line",
      "encoding": {
        "x": {
          "field": "Year",
          "type": "temporal",
          "timeUnit": "year"
        },
        "y": {
          "field": "Mean",
          "type": "quantitative",
          "scale": {
            "zero": false
          },
          "axis": {
            "title": "CO2 in PPM",
            "encode": {
              "title": {
                "fill": {
                  "value": "steelblue"
                }
              }
            },
            "grid": false
          }
        }
      }
    },
    {
      "layer": [
        {
          "data": {
            "values": [
              {
                "rule": 0,
                "_id": 47303
              }
            ]
          },
          "mark": "rule",
          "encoding": {
            "y": {
              "field": "rule",
              "type": "quantitative",
              "axis": {
                "title": "Mean Global Temperature",
                "encode": {
                  "title": {
                    "fill": {
                      "value": "firebrick"
                    }
                  }
                },
                "grid": false
              }
            },
            "opacity": {
              "value": 0.4
            },
            "color": {
              "value": "black"
            }
          }
        },
        {
          "data": {
            "url": "data/temp.csv"
          },
          "mark": "line",
          "encoding": {
            "x": {
              "field": "Year",
              "type": "temporal",
              "timeUnit": "year"
            },
            "y": {
              "field": "Value",
              "type": "quantitative"
            },
            "color": {
              "value": "firebrick"
            }
          }
        }
      ]
    }
  ],
  "resolve": {
    "y": {
      "scale": "independent"
    }
  }
}
```

**Comment by kanitw** [comment]
This plot has 3 axes -- so it is no doubt generating overlapping axes without some smart logic.  
We could modify our axes merging logic to be smarter and add offset (I had TODO in the code), but I don't think it is important to fix for 2.0.

**Comment by kanitw** [comment]
Let's close this since i don't think we'll ever get to fix this.
